### PR TITLE
Add a11y context

### PR DESCRIPTION
**Pull request by estrattonbailey**
Adds a couple a11y props to context so that they're available synchronously in the app. Right now we only check for reduced motion, so I've used that same value from Reanimated as the initial value here (that check IS sync, actually).

This also adds screen reader state for use in Starter Packs and eventually other infinite scroll cases.

Note case on web: screen reader always returns `true`. Since we're primarily concerned about mobile for this immediate need, I've opted to override on web to always be `false`. Eventually we may want to fork these props into something like `isScreenReaderEnabledWeb` and handle some cases differently.

### CODE REVIEW

File: src/state/a11y.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  screenReaderEnabled: false,
+})
+
+export function useA11y() {
+  return React.useContext(Context)
+}
+
+export function Provider({children}: React.PropsWithChildren<{}>) {
+  const [reduceMotionEnabled, setReduceMotionEnabled] = React.useState(
+    isReducedMotion(),
```
Comment (haileyok) [inline src/state/a11y.tsx:18]: nit: should use `() => isReducedMotion()`

File: src/state/a11y.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      enabled => {
+        setScreenReaderEnabled(enabled)
+      },
+    )
+
+    ;(async () => {
+      const [_reduceMotionEnabled, _screenReaderEnabled] = await Promise.all([
+        AccessibilityInfo.isReduceMotionEnabled(),
+        AccessibilityInfo.isScreenReaderEnabled(),
+      ])
```
Comment (haileyok) [inline src/state/a11y.tsx:40]: im planning on adding some methods to get these synchronously in the future to the SwissArmyKnife (and dropping the patch we hacked into `reanimated`), but this seems fine for now.

File: src/state/a11y.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      reduceMotionChangedSubscription.remove()
+      screenReaderChangedSubscription.remove()
+    }
+  }, [])
+
+  const ctx = React.useMemo(() => {
+    return {
+      reduceMotionEnabled,
+      /**
+       * Always returns true on web. For now, we're using this for mobile a11y,
```
Comment (haileyok) [inline src/state/a11y.tsx:55]: really weird there isn't an api for this...maybe a fingerprinting concern ig

**Review by haileyok — APPROVED** [review]
one nit but lgtm

File: src/state/a11y.tsx (hunk 1 above)
Comment (estrattonbailey) [reply src/state/a11y.tsx:18]: Always forget, thanks

**Review by pfrazee — APPROVED** [review]
LGTM

File: src/state/a11y.tsx (hunk 2 above)
Comment (estrattonbailey) [reply src/state/a11y.tsx:40]: Oh awesome. Pry good to have on context anyway for testing, but would be nice to get these sync on first load.

**Pull request merged**